[W5.11a][T16-A2]Ng Shi Kang

ngshikang opened the pull request:
Feature 1: Retrieval of Full Address from Online Geo-Database using Postal Code Input
Add PostalCode class to supplement information on Address, using postal code obtained from input to retrieve full address from online geodirectory (OneMap).
Update Parser and AddCommand classes to incorporate use of this PostalCode class to perform new functionality.
Update user guide in help function.
Update input and expected output for test script.

ngshikang committed:
Added PostalCode class to supplement information on Address, using postal code obtained from input to retrieve full address from online geodirectory.
Updated Parser and AddCommand classes to incorporate use of this PostalCode class to perform new functionality.
Updated user guide in help function.
Updated input and expected output for test script.

asiddharth commented on src/seedu/addressbook/parser/Parser.java:
```diff
@@ -75,33 +76,33 @@ public Command parseCommand(String userInput) {
 
-        case AddCommand.COMMAND_WORD:
-            return prepareAdd(arguments);
+            case AddCommand.COMMAND_WORD:
```
Are there tabs here (and in the following Case statements?)
Please see https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html#layout

Also, the indentation level of the case statements should be the same as that of the switch statements. See point number 10 under layout

asiddharth commented on test/input.txt:
```diff
@@ -106,11 +106,21 @@
+  # should show new list of persons which is now sorted alphabetically
+  sort
+
+
```
It is recommended to use blank lines sparingly as too many blank lines will increase the height of the code, which increases the amount of scrolling.

asiddharth commented on src/seedu/addressbook/data/person/PostalCode.java:
```diff
@@ -0,0 +1,108 @@
+        if (!isValidPostalCode(trimmedPostalCode)) {
+            throw new IllegalValueException(MESSAGE_POSTAL_CODE_CONSTRAINTS);
+        }
+        this.value = trimmedPostalCode;
```
It is recommended to avoid unnecessary use of this with fields.
Please see : https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html#variables (Point number 9)

asiddharth commented on src/seedu/addressbook/commands/AddCommand.java:
```diff
@@ -41,11 +44,23 @@ public AddCommand(String name,
             tagSet.add(new Tag(tagName));
         }
+
+        PostalCode newPostalCode = new PostalCode(postalCode, isPostalCodePrivate);
```
This constructor throws an exception for certain values of postalCode. Should it be in a try-catch block?

asiddharth commented on src/seedu/addressbook/commands/AddCommand.java:
```diff
@@ -41,11 +44,23 @@ public AddCommand(String name,
+                address = newPostalCode.retrieveMatchingAddress();
+            }
+        }
+        catch(IOException e){
```
The form of try-catch block is not correct. Please see https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html#layout (point 11)

asiddharth commented on src/seedu/addressbook/data/person/PostalCode.java:
```diff
@@ -0,0 +1,108 @@
+        }
+        return MESSAGE_ERROR_ADDRESS_NOT_FOUND;
+    }
+
```
Missing header comment. All non trivial methods should have javadoc format header comments.

asiddharth reviewed (commented):
@ngshikang  Some comments added. Please close the PR after reading comments. 
Some more notes : 
1. The Documentation has not been updated (UserGuide.md)
2. As the 1KLoC LO was not completed, you are required to implement 2 features in the 2KLoC codebase.
3. Use the imperative tone in commit messages (ex: "Add tests" instead of "Added tests")
As there is no tutorial this week, you are allowed to complete the above tasks before next week's tutorial.

ngshikang committed:
Edit UserGuide to reflect changes
Edit AddCommand to improve try-catch blocks
Edit PostalCode to remove unnecessary use of "this" and update javadoc header comments
Edit Parser to improve switch case statement indentation
